guided_contribution_tutorial.md

```diff
@@ -0,0 +1,35 @@
+# Guided contribution tutorial
+
+DALI is an open-source project, therefore we are more than happy to accept external contributions (refer to [CONTRIBUTION.md](https://github.com/NVIDIA/DALI/blob/main/CONTRIBUTING.md) for details about regular contributions). Moreover, we are also keen on closer and more structured cooperation. For instance, you can work on a more significant contribution to DALI as a part of your school project, Deep Learning oriented university class, Bachelor's or Master thesis etc... We also recommend such project to be implemented by a group of developers, so if you have any friends who would like to join you, that's even better. This guide outlines the whole process, from scratch to your first merged PR.
+
+1. If you are ready to commit more time to develop DALI, please contact us directly. You can use either Github issues or write to us on [via email](mailto:dali-team@nvidia.com). We will agree upon the contribution goal, make the plan, sign the necessary NDA (Non-disclosure agreement) and establish a more direct communication channel.
+1. I'm assuming, that we have already set up some communication channel (e.g. Slack, mail, etc…). If we didn't yet, please ask your contact point in our team to do so.
+1. You have been assigned with a DALI-team developer, who would help you with everything you need. Let's call him The Mentor. That shall be your primary contact point with us. Nevertheless, feel free to reach out to anyone from the team, we all are happy to help.
+1. Whenever you're in doubt, please reach out to The Mentor. We are here to help you and we are happy to help you. There are problems, that we faced before and we know how to handle them. Reinventing the wheel is not as exciting as solving problems nobody solved before!
+1. Discuss with our team, what would be the topic of your contribution. This part is really important and you shall work closely with The Mentor on the contribution idea. We have some internal documents and process that we would like to keep, and The Mentor is there to help you. Also, The Mentor would contact with any internal stakeholders, to make sure that the contribution is well designed.
```

Comment:
I feel, that the `keep` fits here. I guess we don't expect, that the contributor will fill our internal documents (because he would need the access, which not always can be done). This would need to be done by The Mentor, but of course contributors will assist with this. And the `keep` here is a synonym to `maintain`, or like `keep` in `bookkeeping`. So, `we would like to keep` might be read as `we would like to maintain`, but I just wanted to cruise back from the `maintain` word, since it's so broadly used in our repository, I'm nearly sick of it :)

To conclude, I feel that the `keep` is precisely what we mean in this context.

Reply:
As you wish, but for me in this context the first meaning of keep (to _maintain_ something in our possession) makes this read a bit weird. So what that we keep them internally, we need the contributor to follow some process and help us do some bookkeeping.